src/Proto.Actor/Context/ContextExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="ContextExtensions.cs" company="Asynkron AB">
+//      Copyright (C) 2015-2022 Asynkron AB All rights reserved
+// </copyright>
+// -----------------------------------------------------------------------
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Proto.Context
+{
+    public static class ContextExtensions
+    {
+        /// <summary>
+        /// Calls the callback on token cancellation. If CancellationToken is non-cancellable, this is a noop.
+        /// </summary>
+        public static CancellationTokenRegistration ReenterAfterCancellation(this IContext context, CancellationToken token, Action onCancelled)
```

Comment:
An alternative to having it as an extension is to add it directly to context. Then we could perhaps add cleanup for contexts with callbacks still waiting. With this model that is on the user.